src/middleware.ts

```diff
@@ -0,0 +1,19 @@
+import { NextResponse } from "next/server";
+import { authMiddleware, redirectToSignIn } from "@clerk/nextjs";
+
+export default authMiddleware({
+  publicRoutes: ["/", "/login", "/signup"],
+  afterAuth(auth, req) {
+    if (!auth.userId && !auth.isPublicRoute) {
+      return redirectToSignIn({ returnBackUrl: req.url });
+    }
+    if (auth.userId && !auth.isPublicRoute) {
+      return NextResponse.next();
+    }
+    return NextResponse.next();
```

Comment:
We can add blank lines and remove unnecessary brackets in this function
```suggestion
    if (!auth.userId && !auth.isPublicRoute)
      return redirectToSignIn({ returnBackUrl: req.url });
      
    if (auth.userId && !auth.isPublicRoute)
      return NextResponse.next();
    
    return NextResponse.next();
```